datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeAdapterTest.java

```diff
@@ -300,6 +303,53 @@ void testTwoUsers(WithStorageProvider.StorageDescriptor descriptor) {
 
     }
 
+    @ParameterizedTest
+    @MethodSource("minioOnly")
+    @SneakyThrows
+    void testExhaustedInputStream(WithStorageProvider.StorageDescriptor descriptor) {
+        myinit(descriptor);
+
+        if (!(dfsCredentials instanceof AmazonS3DFSCredentials)) {
+            throw new RuntimeException("programming error");
+        }
+        dfsCredentials = ((AmazonS3DFSCredentials) dfsCredentials).toBuilder().maxConnections(2).requestTimeout(1000).build();
+        mystart();
+// org.bouncycastle.crypto.io.InvalidCipherTextIOException
+
+        String content = "content of document qdm;mwm;ewmfmwemf;we;mfw;emf;llllle";
+        String path = "a/b/c.txt";
+        DSDocument document = new DSDocument(new DocumentFQN(path), new DocumentContent(content.getBytes()));
+        simpleDatasafeService.storeDocument(userIDAuth, document);
+        boolean timeout = false;
+        try {
```

Comment:
Why not to simplify to:

@ParameterizedTest
    @MethodSource("minioOnly")
    @SneakyThrows
    void testExhaustedInputStream(WithStorageProvider.StorageDescriptor descriptor) {
        myinit(descriptor);
        int maxConnections = 2;

        if (!(dfsCredentials instanceof AmazonS3DFSCredentials)) {
            throw new RuntimeException("programming error");
        }
        dfsCredentials = ((AmazonS3DFSCredentials) dfsCredentials).toBuilder()
                .maxConnections(maxConnections)
                .requestTimeout(1000)
                .build();
        mystart();

        String content = "content of document qdm;mwm;ewmfmwemf;we;mfw;emf;llllle";
        String path = "a/b/c.txt";
        DSDocument document = new DSDocument(new DocumentFQN(path), new DocumentContent(content.getBytes()));
        simpleDatasafeService.storeDocument(userIDAuth, document);

        // create more `not closed requests` than pool can handle
        IntStream.range(0, maxConnections + 1).forEach(it -> {
            DSDocumentStream ds = simpleDatasafeService.readDocumentStream(userIDAuth, document.getDocumentFQN());
            try {
                // This causes exceptions when using authenticated encryption
                ds.getDocumentStream().close();
            } catch (Exception ex) {
                // Ignoring exception, i'm badly behaved client...
            }
        });
    }

Reply:
yep